tools/rpkg/R/register.R

```diff
@@ -36,3 +36,27 @@ duckdb_unregister <- function(conn, name) {
   invisible(TRUE)
 }
 
+#' @rdname duckdb_register
+#' @export
+duckdb_register_arrow <- function(conn, name, arrow_table) {
+  stopifnot(dbIsValid(conn))
+  record_batch_reader <- arrow::Scanner$create(arrow_table)$ToRecordBatchReader()
+
+    # create a function to pass to c-land
```

Comment:
Why is this an R function you pass to C? Why not do this here and just pass `stream_ptr` to `duckdb_register_arrow_R`?

Reply:
Because we need to be able to re-create the stream ptr in case we have multiple reads on this source.